File: {{cookiecutter.project_slug}}/src/{{cookiecutter.project_slug}}/task_data_management.py
import pytask
from {{cookiecutter.project_slug}}.config import BLD
from {{cookiecutter.project_slug}}.config import SRC
{% if cookiecutter.example_language == 'python' %}
import pandas as pd
from {{cookiecutter.project_slug}}.data_management import clean_data
from {{cookiecutter.project_slug}}.utilities import read_yaml

@pytask.mark.depends_on({"data_info": SRC / "data_management" / "data_info.yaml"})
@pytask.mark.produces(BLD / "data" / "data.csv")
def task_get_data(depends_on, produces):
    data_info = read_yaml(depends_on["data_info"])
    data = pd.read_csv(data_info["url"])
    data.to_csv(produces, index=False)


@pytask.mark.depends_on(
    {
        "data_info": SRC / "data_management" / "data_info.yaml",
        "data": BLD / "data" / "data.csv",
    }
)
@pytask.mark.produces(BLD / "data" / "data_clean.csv")
def task_clean_data(depends_on, produces):
    data_info = read_yaml(depends_on["data_info"])
    data = pd.read_csv(depends_on["data"])
    data = clean_data(data, data_info)
    data.to_csv(produces, index=False)
{% endif %}

{% if cookiecutter.example_language == 'r' %}
@pytask.mark.r(script=SRC / "task_data_management.r", serializer="yaml")
@pytask.mark.task(kwargs={"task": "get_data"})
@pytask.mark.depends_on({"data_info": SRC / "data_management" / "data_info.yaml"})
@pytask.mark.produces(BLD / "data" / "data.csv")
def task_get_data():
    pass


@pytask.mark.r(script=SRC / "task_data_management.r", serializer="yaml")
@pytask.mark.task(kwargs={"task": "clean_data"})
@pytask.mark.depends_on(
    {
        "data_info": SRC / "data_management" / "data_info.yaml",
        "data": BLD / "data" / "data.csv",
    }
)
@pytask.mark.produces(BLD / "data" / "data_clean.csv")
def task_clean_data():
    pass
{% endif %}

File: {{cookiecutter.project_slug}}/src/task_data_management.R
library(yaml)

# ======================================================================================
# Tasks
# ======================================================================================

task_get_data = function(depends_on, produces) {
  data_info = yaml::yaml.load_file(depends_on[["data_info"]])
  data = read.csv(data_info[["url"]])
  write.csv(data, file=produces, col.names=TRUE)
}


task_clean_data = function(depends_on, produces) {
  data_info = yaml::yaml.load_file(depends_on[["data_info"]])
  data = read.csv(data_info[["data"]])
  data = clean_data(data, data_info)
  write.csv(data, file=produces, col.names=TRUE)
}

# ======================================================================================
# Main
# ======================================================================================

args <- commandArgs(trailingOnly=TRUE)
path_to_json <- args[length(args)]
config <- yaml::yaml.load_file(path_to_json)

produces = config[["produces"]]
depends_on = config[["depends_on"]]
task = config[["task"]]

if (task == "get_data") {

  task_get_data(produces, depends_on)

} else if (task == "clean_data") {

  source(file.path(config[["SRC"]], "data_management", "clean_data.R"))  
  task_clean_data(depends_on, produces)

} else {
  stop("task needs to be either 'get_data' or 'clean_data'.")
}